src/cg.rs

```diff
@@ -27,14 +28,13 @@ impl CGImpl {
             let subview: id = NSView::alloc(nil).initWithFrame_(NSView::frame(view));
             layer.set_contents_gravity(ContentsGravity::TopLeft);
             layer.set_needs_display_on_bounds_change(false);
-            layer.set_contents_scale(window.backingScaleFactor());
             subview.setLayer(layer.id());
             subview.setAutoresizingMask_(NSViewWidthSizable | NSViewHeightSizable);
 
             view.addSubview_(subview); // retains subview (+1) = 2
             let _: () = msg_send![subview, release]; // releases subview (-1) = 1
         }
-        Ok(Self { layer })
+        Ok(Self { layer, window })
```

Comment:
Wrt. memory management: `AppKitWindowHandle` should ideally guarantee that the window is alive, but I think I'd prefer a `let window: id = msg_send![window, retain];` before we store it, and a `let _: () = msg_send![self.window, release];` in `CGImpl`'s `Drop`.

Reply:
Thanks, that's a good idea. Done.